components/proxy/src/test/kotlin/com/hotels/styx/proxy/backends/file/FileBackedBackendServicesRegistryTest.kt

```diff
@@ -0,0 +1,217 @@
+import com.hotels.styx.infrastructure.configuration.json.ObjectMappers.addStyxMixins
+import io.kotlintest.matchers.collections.shouldContain
+import io.kotlintest.matchers.collections.shouldContainAll
+import io.kotlintest.specs.DescribeSpec
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import java.io.File
+import java.nio.file.Paths
+import java.util.concurrent.TimeUnit.MILLISECONDS
+import java.util.concurrent.TimeUnit.SECONDS
+
+
+class FileBackedBackendServicesRegistryKotlinTest: DescribeSpec() {
+    val Mapper = addStyxMixins(ObjectMapper(YAMLFactory()))
+
+    val backendServiceOne = BackendService.Builder()
+            .id("webapp")
+            .origins(origin("webapp", "webapp-02", "localhost", 9091), origin("webapp", "webapp-01", "localhost", 9090))
+            .connectionPoolConfig(ConnectionPoolSettings.Builder()
+                    .connectTimeout(8000, MILLISECONDS)
+                    .maxConnectionsPerHost(300)
+                    .maxPendingConnectionsPerHost(50)
+                    .build())
+            .healthCheckConfig(newHealthCheckConfigBuilder()
+                    .uri("/version.txt")
+                    .interval(23456)
+                    .timeout(2000)
+                    .healthyThreshold(3)
+                    .unhealthyThreshold(5)
+                    .build())
+            .stickySessionConfig(StickySessionConfig.Builder()
+                    .enabled(true)
+                    .timeout(86400, SECONDS)
+                    .build())
+            .build()
+
+
+    fun origin(appId: String = "", id: String, host: String, port: Int): Origin {
+        return newOriginBuilder(host, port)
+                .applicationId(appId)
+                .id(id)
+                .build()
+    }
+
+    fun writeToFile(applications: List<BackendService>, path: String): File {
+        val output = File(createOnMissing(path))
+        Mapper.writer().writeValue(output, applications)
+        return output
+    }
+
+    fun createOnMissing(path: String): String {
+        val absolutePath = Paths.get(javaClass.getResource("/").toURI()).resolve(path).toFile()
+        if (!absolutePath.getParentFile().exists()) {
+            absolutePath.getParentFile().mkdirs()
+        }
+        return absolutePath.toString()
+    }
+
+
+    init {
+        describe("A file backed backend services registry") {
+            context("reading yaml content") {
+                it("should read backend services configured in the yaml file") {
+                    val registry = FileBackedBackendServicesRegistry.create("classpath:conf/origins/origins-for-jsontest.yml")
+
+                    StyxFutures.await(registry.start())
+                    val backendServices = registry.get()
+
+                    assert(backendServices.toList().size == 2)
+                    assert(backendServices.first() == backendServiceOne)
+
+                    StyxFutures.await(registry.stop())
```

Comment:
do we need to wait for the stop?  I suppose that means we cannot run the tests in parallel, it would be nice to improve that...

Reply:
Okay. I'll keep this in mind. This problem is inherited from the original Scala test. So it isn't introduced by this PR.